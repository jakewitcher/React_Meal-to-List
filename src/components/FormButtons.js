import React from 'react';
import { Link } from 'react-router-dom';

const FormButtons = () => {
    return (
        <div className="form-tabs">
            <Link to="/meals">
                <button className="form-tabs__button form-tabs__button--meal">New meal</button>
            </Link>
            <Link to="/grocerylists">
                <button className="form-tabs__button form-tabs__button--grocery">New grocery list</button>
            </Link>
        </div>
    );
};

export default FormButtons;