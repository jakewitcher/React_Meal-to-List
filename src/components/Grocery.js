import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { onDeleteGrocery } from "../store/grocery/actions";

export const Grocery = props => {
  const { name, id, itemList } = props;
  return (
    <div className="list-type">
      <Link className="list-type__link" to={`grocerylists/grocery-list/${id}`}>
        <p className="list-type__name">{name}</p>
      </Link>
      <p className="list-type__details">
        number of items:{" "}
        <span className="list-type__details list-type__details--span">
          {itemList.length}
        </span>
      </p>
      <div className="list-type__update">
        <Link className="list-type__link" to={`grocerylists/edit/${id}`}>
          <i className="list-type__edit fas fa-edit" />
        </Link>
        <i
          className="list-type__delete fas fa-trash"
          onClick={() => props.onDeleteGrocery(id)}
        />
      </div>
    </div>
  );
};

const mapDispatchToProps = dispatch => {
  return {
    onDeleteGrocery: id => dispatch(onDeleteGrocery(id))
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(Grocery);
