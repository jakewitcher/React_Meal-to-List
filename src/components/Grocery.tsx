import React from "react";
import { Dispatch } from "redux";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { onDeleteGrocery } from "../store/grocery/actions";
import { Item } from "../store/item/types";

interface IGroceryProps {
  name: string;
  id: string;
  itemList: Item[];
  onDeleteGrocery(id: string): void;
}

export const Grocery: React.FC<IGroceryProps> = props => {
  const { name, id, itemList } = props;
  return (
    <div className="list-type">
      <Link className="list-type__link" to={`grocerylists/grocery-list/${id}`}>
        <p className="list-type__name">{name}</p>
      </Link>
      <p className="list-type__details">
        number of items:{" "}
        <span className="list-type__details list-type__details--span">
          {itemList.length}
        </span>
      </p>
      <div className="list-type__update">
        <Link className="list-type__link" to={`grocerylists/edit/${id}`}>
          <i className="list-type__edit fas fa-edit" />
        </Link>
        <i
          className="list-type__delete fas fa-trash"
          onClick={() => props.onDeleteGrocery(id)}
        />
      </div>
    </div>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    onDeleteGrocery: (id: string) => dispatch(onDeleteGrocery(id))
  };
};

export default connect(
  undefined,
  mapDispatchToProps
)(Grocery);
