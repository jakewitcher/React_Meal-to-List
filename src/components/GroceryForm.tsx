import React, { ChangeEvent } from "react";
import { Item } from "../store/item/types";
import { Meal, MealState } from "../store/meal/types";
import { Grocery, GroceryMeals } from "../store/grocery/types";

interface IMealFormProps {
  groceryName: string;
  itemList: [] | Item[];
  groceryId: string;
  selectedMeal: Meal;
  meals: MealState;
  groceryListMeals: GroceryMeals;
  addMealToList(): void;
  selectedMealChange(e: ChangeEvent<HTMLSelectElement>): void;
  groceryNameChange(e: React.FormEvent<HTMLInputElement>): void;
  resetGrocery(): void;
  updateGrocery(grocery: Grocery): void;
}

const GroceryForm: React.FC<IMealFormProps> = props => {
  const {
    selectedMealChange,
    selectedMeal,
    meals,
    groceryName,
    itemList,
    groceryListMeals,
    groceryNameChange,
    addMealToList,
    resetGrocery,
    updateGrocery,
    groceryId
  } = props;

  const handleGroceryChange = (
    name: string,
    itemList: Item[],
    groceryListMeals: GroceryMeals,
    id: string
  ) => {
    updateGrocery({
      name,
      itemList,
      groceryListMeals,
      id
    });
    resetGrocery();
  };

  return (
    <div className="form--grocery">
      <div className="input-box input-box--meal-name">
        <p>Meals</p>
        <select
          className="input input--dropdown"
          name="selectedMeal"
          value={selectedMeal.name}
          onChange={selectedMealChange}
        >
          {meals.mealList.map(meal => (
            <option value={meal.name} key={meal.id}>
              {meal.name}
            </option>
          ))}
        </select>
      </div>
      <div className="button-box__add--meal">
        <button className="button button__add" onClick={addMealToList}>
          Add Meal
        </button>
      </div>
      <div className="input-box__grocery-name">
        <p>Grocery List Name</p>
        <input
          className="input__grocery-name"
          type="text"
          value={groceryName}
          onChange={groceryNameChange}
        />
      </div>
      <div className="button-box__save--grocery">
        <button
          className="button button__save"
          onClick={() =>
            handleGroceryChange(
              groceryName,
              itemList,
              groceryListMeals,
              groceryId
            )
          }
        >
          Save Grocery List
        </button>
      </div>
    </div>
  );
};

export default GroceryForm;
