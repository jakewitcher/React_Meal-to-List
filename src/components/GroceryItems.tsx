import React from "react";
import GroceryItem from "./GroceryItem";
import { Item } from "../store/item/types";

interface IGroceryItemsProps {
  isDetails: boolean;
  itemList: Item[];
  groceryName: string;
  deleteGroceryItem(name: string): void;
}

const GroceryItems: React.FC<IGroceryItemsProps> = props => {
  const { groceryName, itemList, deleteGroceryItem, isDetails } = props;
  return (
    <div className="itemlist">
      <h1 className="itemlist__header">{groceryName || "New Grocery List"}</h1>
      {itemList.length > 0 && (
        <ul className="itemlist__body">
          {itemList.map(item => {
            return (
              <GroceryItem
                item={item}
                deleteGroceryItem={deleteGroceryItem}
                key={item.itemName}
                isDetails={isDetails}
              />
            );
          })}
        </ul>
      )}
    </div>
  );
};

export default GroceryItems;
