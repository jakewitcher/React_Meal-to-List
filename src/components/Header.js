import React from 'react';

const Header = () => {
    return (
        <div className="header">
            <h1 className="header__title">Meal to List</h1>
            <hr />
            <h3 className="header__subtitle">Meal Planning Made Easy</h3>
        </div>
    );
};

export default Header;
