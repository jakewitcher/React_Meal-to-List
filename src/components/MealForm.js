import React from "react";

const MealForm = props => {
  const {
    mealNameChange,
    itemNameChange,
    amountChange,
    unitChange,
    addItem,
    resetMeal,
    mealName,
    itemName,
    amount,
    unit,
    itemList,
    updateMeal,
    mealId
  } = props;

  const pluralUnitValue = (amount, unitType) => {
    if (amount < 2) {
      return `${unitType}`;
    }
    return `${unitType}s`;
  };

  const handleMealChange = (name, id, itemList) => {
    updateMeal({
      name,
      itemList,
      id
    });
    resetMeal();
  };

  return (
    <div className="form">
      <div className="input-box input-box--item-name">
        <p>Item</p>
        <input
          className="input"
          type="text"
          value={itemName}
          onChange={itemNameChange}
        />
      </div>

      <div className="input-box input-box--amount">
        <p>Amount</p>
        <input
          className="input"
          type="number"
          value={amount}
          onChange={amountChange}
        />
      </div>

      <div className="input-box input-box--unit">
        <p>Unit</p>
        <select
          className="input input--dropdown"
          name="unit"
          value={unit}
          onChange={unitChange}
        >
          <option value={""}>{"none"}</option>
          <option value={pluralUnitValue(amount, "pound")}>
            {pluralUnitValue(amount, "pound")}
          </option>
          <option value={pluralUnitValue(amount, "ounce")}>
            {pluralUnitValue(amount, "ounce")}
          </option>
          <option value={pluralUnitValue(amount, "gallon")}>
            {pluralUnitValue(amount, "gallon")}
          </option>
          <option value={pluralUnitValue(amount, "quart")}>
            {pluralUnitValue(amount, "quart")}
          </option>
          <option value={pluralUnitValue(amount, "pint")}>
            {pluralUnitValue(amount, "pint")}
          </option>
          <option value={pluralUnitValue(amount, "item")}>
            {pluralUnitValue(amount, "item")}
          </option>
          <option value={pluralUnitValue(amount, "bag")}>
            {pluralUnitValue(amount, "bag")}
          </option>
          <option value={pluralUnitValue(amount, "container")}>
            {pluralUnitValue(amount, "container")}
          </option>
        </select>
      </div>
      <div className="button-box__add">
        <button className="button button__add" onClick={addItem}>
          Add Item
        </button>
      </div>
      <div>
        <div className="input-box__meal-name">
          <p>Meal Name</p>
          <input
            className="input__meal-name"
            type="text"
            value={mealName}
            onChange={mealNameChange}
          />
        </div>
        <div className="button-box__save">
          <button
            className="button button__save"
            onClick={() => handleMealChange(mealName, mealId, itemList)}
          >
            Save Meal
          </button>
        </div>
      </div>
    </div>
  );
};

export default MealForm;
