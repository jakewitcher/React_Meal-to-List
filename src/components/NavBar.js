import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { onStartLogout } from '../actions/auth';

export class NavBar extends Component {
    state = {
        navLinkClassName: 'navbar__link',
        isToggled: true,
    }

    toggleNav = () => {
        let { isToggled } = this.state;
        this.setState({ isToggled: !isToggled });
        isToggled ? this.setState({ navLinkClassName: 'navbar__link--show' }) : this.setState({ navLinkClassName: 'navbar__link' });
    }

    render() {
        return (
            <div className="navbar">
                <NavLink to="/" className="navbar__brand" activeClassName="navbar__link--active" exact={true}>
                    <p>Meal to List</p>
                </NavLink>
                <i onClick={this.toggleNav} className="fas fa-bars navbar__hamburger" />
                <NavLink to="/meals" className={this.state.navLinkClassName} activeClassName="navbar__link--active" exact={true}>
                    <p>Meals</p>
                </NavLink>
                <NavLink to="/meals/create" className={this.state.navLinkClassName} activeClassName="navbar__link--active" exact={true}>
                    <p>Create a New Meal</p>
                </NavLink>
                <NavLink to="/grocerylists" className={this.state.navLinkClassName} activeClassName="navbar__link--active" exact={true}>
                    <p>Grocery Lists</p>
                </NavLink>
                <NavLink to="/grocerylists/create" className={this.state.navLinkClassName} activeClassName="navbar__link--active" exact={true}>
                    <p>Create a New Grocery List</p>
                </NavLink>
                <button className={`navbar__link-button ${this.state.navLinkClassName}`} onClick={this.props.onStartLogout}>Logout</button>
            </div>
        );
    }
};

const mapDispatchToProps = (dispatch) => ({
    onStartLogout: () => dispatch(onStartLogout())
});

export default connect(undefined, mapDispatchToProps)(NavBar);