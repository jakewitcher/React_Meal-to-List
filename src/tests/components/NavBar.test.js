import React from 'react';
import { shallow } from 'enzyme';
import { NavBar } from '../../components/NavBar';

test('should render NavBar correctly', () => {
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
});