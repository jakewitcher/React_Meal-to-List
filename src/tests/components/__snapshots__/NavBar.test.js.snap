// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavBar correctly 1`] = `
<div
  className="navbar"
>
  <NavLink
    activeClassName="navbar__link--active"
    aria-current="page"
    className="navbar__brand"
    exact={true}
    to="/"
  >
    <p>
      Meal to List
    </p>
  </NavLink>
  <i
    className="fas fa-bars navbar__hamburger"
    onClick={[Function]}
  />
  <NavLink
    activeClassName="navbar__link--active"
    aria-current="page"
    className="navbar__link"
    exact={true}
    to="/meals"
  >
    <p>
      Meals
    </p>
  </NavLink>
  <NavLink
    activeClassName="navbar__link--active"
    aria-current="page"
    className="navbar__link"
    exact={true}
    to="/meals/create"
  >
    <p>
      Create a New Meal
    </p>
  </NavLink>
  <NavLink
    activeClassName="navbar__link--active"
    aria-current="page"
    className="navbar__link"
    exact={true}
    to="/grocerylists"
  >
    <p>
      Grocery Lists
    </p>
  </NavLink>
  <NavLink
    activeClassName="navbar__link--active"
    aria-current="page"
    className="navbar__link"
    exact={true}
    to="/grocerylists/create"
  >
    <p>
      Create a New Grocery List
    </p>
  </NavLink>
  <button>
    Logout
  </button>
</div>
`;
